### PR TITLE
Sample details refactor/ sample analyses

### PR DESCRIPTION
**Pull request by deepsidhu85**
## Description of changes
What did you change in this pull request?  Provide a description of files changed, user interactions changed, etc.  Include how to test your changes.

Added functionality to display analyses ran for a sample (admin can view all analyses ran for a sample and every other user can only view analyses that they have ran)

sample-analyses

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

~* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~
* [ ] Tests added (or description of how to test) for any new features.
~* [ ] User documentation updated for UI or technical changes.~

### CODE REVIEW

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UISampleService.java [hunk 1]
```diff
@@ -81,13 +88,18 @@
...
 
 	@Autowired
 	public UISampleService(SampleService sampleService, ProjectService projectService,
 			UpdateSamplePermission updateSamplePermission, SequencingObjectService sequencingObjectService,
 			GenomeAssemblyService genomeAssemblyService, MessageSource messageSource, UICartService cartService,
 			MetadataTemplateService metadataTemplateService, MetadataEntryRepository metadataEntryRepository,
-			MetadataRestrictionRepository metadataRestrictionRepository) {
+			MetadataRestrictionRepository metadataRestrictionRepository,
+			AnalysisSubmissionRepository analysisSubmissionRepository, UserService userService,
+			IridaWorkflowsService iridaWorkflowsService) {
```
Comment (joshsadam) [inline src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UISampleService.java:102]: This services is getting really big, do you think we should move this stuff to maybe an UIAnalysesService?

File: src/main/webapp/resources/js/components/samples/components/SampleAnalyses.jsx [hunk 2]
```diff
@@ -1,43 +1,153 @@
...
+
+  /*
+  On page load get the sample analyses
+   */
+  React.useEffect(() => {
+    fetchSampleAnalyses({ sampleId: sample.identifier })
+      .then((analysesList) => {
+        dispatch(setSampleAnalyses({ analyses: analysesList }));
+      })
+      .catch((error) => {});
```
Comment (joshsadam) [inline src/main/webapp/resources/js/components/samples/components/SampleAnalyses.jsx:47]: Are you doing anything with this error?

File: src/main/webapp/resources/js/components/samples/components/SampleAnalyses.jsx [hunk 3]
```diff
@@ -1,43 +1,153 @@
...
   const columns = [
     {
-      title: i18n("SampleAnalyses.sampleId"),
-      key: "sample_id",
+      title: i18n("SampleAnalyses.analysisSubmissionName"),
+      dataIndex: "name",
+      key: "name",
+      render(name, data) {
+        return (
+          <Paragraph ellipsis={{ rows: 1 }}>
```
Comment (joshsadam) [inline src/main/webapp/resources/js/components/samples/components/SampleAnalyses.jsx:58]: This actually works with children? I thought the text had to be a direct child.  Test it out, if it does not work, you might be able to next the paragraph inside the anchor tag

File: src/main/webapp/resources/js/components/samples/components/SampleAnalyses.jsx [hunk 4]
```diff
@@ -1,43 +1,153 @@
...
+      ),
     },
     {
-      title: i18n("SampleAnalyses.analysisType"),
-      key: "analysis_type",
+      title: i18n("SampleAnalyses.analysisState"),
+      dataIndex: "state",
+      key: "analysisType",
+      render: (state) => (
+        <span className="t-analysis-state">
```
Comment (joshsadam) [inline src/main/webapp/resources/js/components/samples/components/SampleAnalyses.jsx:92]: Make this a component

**Review by joshsadam — CHANGES_REQUESTED** [review]
Just a few changes, overall I think it is looking good.

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UISampleService.java (hunk 1 above)
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UISampleService.java:98]: Yup good idea! Moved to new service in d3e0984150f4ca77d8fd65841d0c7d45ece15f41

File: src/main/webapp/resources/js/components/samples/components/SampleAnalyses.jsx (hunk 2 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/samples/components/SampleAnalyses.jsx]: Was not being used for anything. Removed in 3cc32f03e78bb36dfc7aba5f0f1fa4d2fe94c0a6

File: src/main/webapp/resources/js/components/samples/components/SampleAnalyses.jsx (hunk 4 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/samples/components/SampleAnalyses.jsx]: Created component in 77a12d1d1497c2e6253bb09c7427afef56539876

File: src/main/webapp/resources/js/components/samples/components/SampleAnalyses.jsx (hunk 3 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/samples/components/SampleAnalyses.jsx]: Removed paragraph and set ellipsis: true and width for column in 84b01a29dc2ced20afe8e58598ff464cf93399bb

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UISampleService.java (hunk 1 above)
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UISampleService.java:98]: Awesome, thanks

File: src/main/webapp/resources/js/components/samples/components/SampleAnalyses.jsx [hunk 5]
```diff
@@ -1,43 +1,133 @@
...
+        <Search
+          placeholder={i18n("SampleAnalyses.inputSearchText")}
+          onChange={(e) => searchSubmissions(e.target.value)}
+          allowClear={true}
+          className="t-sample-analyses-search-input"
+        />
+      </Col>
+      <Col span={24}>
+        <Table
+          bordered
```
Comment (joshsadam) [inline src/main/webapp/resources/js/components/samples/components/SampleAnalyses.jsx:121]: Try removing this and see how it looks.  I don't think we render a lot of bordered tables, but if it adds somethine you feel is important let me know and we can keep it.

**Pull request merged**